### PR TITLE
32866: Updated php-amqplib/php-amqplib to the latest version 3.0.0

### PR DESCRIPTION
**Pull request by anzin**
### Description (*)

I've updated php-amqplib/php-amqplib to the latest version 3.0.0

list usages the library in the codebase:

\Magento\AmqpStore\Plugin\AsynchronousOperations\MassConsumerEnvelopeCallback **compatible**
\Magento\AmqpStore\Plugin\Framework\Amqp\Bulk\Exchange **compatible**
\Magento\Framework\Amqp\Config **compatible**
\Magento\Framework\Amqp\Exchange **compatible**
\Magento\Framework\Amqp\Queue **not compatible, in 3.0.0 removed  PhpAmqpLib\Exception\AMQPProtocolConnectionException**
\Magento\Framework\Amqp\Bulk\Exchange **compatible**
\Magento\Framework\Amqp\Connection\Factory **compatible**
\Magento\Framework\Amqp\Test\Unit\TopologyInstallerTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Bulk\ExchangeTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Connection\FactoryTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Topology\BindingInstallerTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Topology\ExchangeInstallerTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Topology\QueueInstallerTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Topology\BindingInstallerType\ExchangeTest **compatible**
\Magento\Framework\Amqp\Test\Unit\Topology\BindingInstallerType\QueueTest **compatible**
\Magento\Framework\Amqp\Topology\BindingInstaller **compatible**
\Magento\Framework\Amqp\Topology\BindingInstallerInterface **compatible**
\Magento\Framework\Amqp\Topology\ExchangeInstaller **compatible**
\Magento\Framework\Amqp\Topology\QueueInstaller **compatible**
\Magento\Framework\Amqp\Topology\BindingInstallerType\Exchange **compatible**
\Magento\Framework\Amqp\Topology\BindingInstallerType\Queue **compatible**
\Magento\Framework\MessageQueue\Bulk\Rpc\Publisher **compatible**

###  Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/567

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32866

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by anzin** [comment]
@magento run all tests

**Comment by anzin** [comment]
@magento run Unit Tests, Functional Tests B2B

**Review by eduard13 — APPROVED** [review]

File: lib/internal/Magento/Framework/Amqp/Queue.php [hunk 1]
```diff
@@ -73,11 +76,11 @@ public function dequeue()
         /** @var AMQPMessage $message */
         try {
             $message = $channel->basic_get($this->queueName);
-        } catch (AMQPProtocolConnectionException $e) {
+        } catch (AMQPTimeoutException $exception) {
```
Comment (Den4ik) [inline lib/internal/Magento/Framework/Amqp/Queue.php]: As I see `basic_get` could throw not only described in phpdoc `AMQPTimeoutException` exception. I believe that we should use `Exception` here

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @anzin 
Thanks for your contribution. Please look at my comment

**Comment by anzin** [comment]
@magento run all tests

**Review by eduard13 — CHANGES_REQUESTED** [review]
@anzin, could you please create a similar PR in EE repo?
Thank you.

**Comment by anzin** [comment]
Hello @eduard13 I created a similarly PR in EE repo: https://github.com/magento/partners-magento2ee/pull/567

Thank you.

**Comment by anzin** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run all tests

**Comment by anzin** [comment]
@magento run Unit Tests , Functional Tests B2B, Functional Tests EE

**Comment by Den4ik** [comment]
Changes looks good for me. Failed tests should not be related to changes. But waiting current rerun

**Comment by andrewbess** [comment]
@magento run all tests

**Review by eduard13 — APPROVED** [review]

**Comment by andrewbess** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests EE

**Comment by andrewbess** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Review by Den4ik — APPROVED** [review]
Failed tests looks as not related to changes

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Comment by andrewbess** [comment]
Unambiguously, failed tests is not related to changes.
But it will be merged only after passing all tests.

**Comment by andrewbess** [comment]
@magento run Functional Tests B2B

**Review by andrewbess — APPROVED** [review]
Hello @anzin 
Thank you for your contribution.
Great job

**Comment by andrewbess** [comment]
Currently, service currconv.com is not available.
service-currconv-com-not-available
There is only one error in functional tests.
So, this PR and related can be merged.

**Pull request merged**